src/engraving/dom/engravingitem.cpp

```diff
@@ -753,6 +755,8 @@ PointF EngravingItem::canvasPos() const
             measure = toSegment(explicitParent()->explicitParent())->measure();
         } else if (explicitParent()->isFretDiagram()) {
             return p + parentItem()->canvasPos();
+        } else if (explicitParent()->isBarLine()) {
+            return p + parentItem()->canvasPos();
```

Comment:
Removing the ON_STAFF flag these two exceptions for pagePos and canvasPos shouldn't be needed anymore